src/common/evtloopcmn.cpp

```diff
@@ -254,95 +345,17 @@ int wxEventLoopManual::DoRun()
 #if wxUSE_EXCEPTIONS
     for ( ;; )
     {
+        if ( wxSystemOptions::IsFalse("catch-unhandled-exceptions") )
+        {
+            Loop();
+            // exit the outer loop as well
+            break;
+        }
+        else
```

Comment:
This looks weird to me, why do we put this inside a loop only to break out of it unconditionally? Why not just put

```cpp
if ( ...propagate exceptions... )
{
    Loop();
    return m_exitcode;
}
```

?

Reply:
Can be done.